infor-design/enterprise
# Misc - Add datagrid patches for 4.31 [v4.31.x]

**Explain the _details_ for making this change. What existing problem does the pull request solve?**

Fixed some issues reported as urgent by landmark.

1) Added ability to scroll left side of a frozen tree
2) Fixed alignment issues if the selection checkbox column
3) Fixed editor alignment issues in all themes (lookup, datepicker, dropdown were worst)
4) Fixed some small doc errors

Please review but i will need to update some images.

**Related github/jira issue (required)**:
Fixes #3852
Fixes #4266

**Steps necessary to review your pull request (required)**:
**Issue One**
- go to http://localhost:4000/components/datagrid/example-multiselect
- check that the checkbox column for selection is now centered vertically in the cell
- test all 8 themes / row heights

**Issue Two**
- go to http://localhost:4000/components/datagrid/test-editable-lookup.html
- click in each cell and edit
- check that the editor looks decent and is usable
- test all 8 themes / row heights
- go to http://localhost:4000/components/datagrid/example-editable
- click in each cell and edit
- check that the editor looks decent and is usable
- test all 8 themes / row heights
- go to http://localhost:4000/components/datagrid/test-lookup-cell-overlap.html
- click in each cell and edit
- check that the editor looks decent and is usable
- test all 8 themes / row heights

**Issue Three**
- go to http://localhost:4000/components/datagrid/test-tree-frozen-columns.html
- notice you can expand the tree and scroll the left side now
- re check http://localhost:4000/components/datagrid/example-frozen-columns.html -> ensure no significant changes

**Included in this Pull Request**:
